browser-test/src/admin/admin_api_keys.test.ts

```diff
@@ -17,23 +21,47 @@ test.describe('Managing API keys', () => {
       await validateScreenshot(page, 'new-api-key-no-programs')
     })
 
-    await test.step('Add and publish program', async () => {
+    await test.step('Add external program only', async () => {
+      await enableFeatureFlag(page, 'external_program_cards_enabled')
+      await adminPrograms.addExternalProgram(
+        externalProgramName,
+        externalProgramDescription,
+        'https://external.gov',
+        ProgramVisibility.PUBLIC,
+      )
+    })
+
+    await test.step('Validate new api key page still shows no programs', async () => {
+      await adminApiKeys.gotoNewApiKeyPage()
+      await expect(page.getByText(externalProgramName)).toBeHidden()
+      await expect(
+        page.getByText(
+          'You must create and publish a program before creating an API Key',
+        ),
+      ).toBeVisible()
+    })
+
+    await test.step('Add and publish default program', async () => {
       await adminPrograms.addProgram(
-        programName,
-        programDescription,
+        internalProgramName,
+        internalProgramDescription,
         'https://usa.gov',
       )
-
       await adminPrograms.publishAllDrafts()
     })
 
-    await test.step('Validate new api key page', async () => {
+    await test.step('Validate external programs do not appear in API key form', async () => {
       await adminApiKeys.gotoNewApiKeyPage()
-      await validateScreenshot(page, 'new-api-key-page')
+
+      await expect(page.getByText('api-using-program')).toBeVisible()
+      await expect(page.getByText(externalProgramName)).toBeHidden()
```

Comment:
this should be `external-program`, and not the program name (since the program slug is what it's used). Also, it's preferred to retrieve elements by `getByRole()`. Lets do it separately since the PR was already submitted